tor/core/validation.py

```diff
@@ -16,34 +26,51 @@ def _header_check(reply, config, tor_link=ToR_link):
 
 
 def _thread_title_check(original_post, history_item):
-    # Verify that the link titles match. On the original post, it will be
-    # removed, but we should still be able to extract the title of the
-    # submission it's on. Then we check to see if the title for that submission
-    # is in the r/ToR post, mirroring the max line truncation that's in
-    # posts.py.
+    """
+    Verify that the link titles match. On the original post, it will be
+    removed, but we should still be able to extract the title of the
+    submission it's on. Then we check to see if the title for that submission
+    is in the r/ToR post, mirroring the max line truncation that's in
+    posts.py.
+
+    :param original_post: Comment object; comment that says "done".
+    :param history_item: Comment object; comment pulled from user's history.
+    """
     max_title_length = 250
     return (
         history_item.link_title[:max_title_length - 4] in
         original_post.link_title
     )
 
 
+def _thread_author_check(original_post, history_item, config):
+    return (
+        history_item.submission.author == config.r.submission(
+            url=original_post.submission.url
+        ).author
+    )
+
+
 def _author_history_check(post, config):
     """
     Pull the five latest items from the user's history. Chances are that's
     enough to see if they've actually done the post or not without slowing
     everything down _too_ much. See if any of those five items look right
     and complete the post if it's the transcript we're looking for.
 
+    Warning: this is not very fast, but it does the job. Definitely something
+    we're going to have to circle back to when we separate out the jobs.
+
     :param post: The Comment object that contains the string 'done'.
     :param config: the global config object.
     :return: True if the post is found in the history, False if not.
     """
     for history_post in post.author.new(limit=5):
         if (
-            _header_check(history_post, config) and
             history_post.is_root and
-            _thread_title_check(post, history_post)
+            _footer_check(history_post, config) and
+            _thread_title_check(post, history_post) and
+            _thread_author_check(post, history_post, config)
```

Comment:
I don't recall, does python evaluate these chained conditionals like a binary operator? Something like `a and b and c` where it does not evaluate `c` if `b` is false?

Just trying to separate out ruby and python knowledge. Ruby had some concerns in that area with using `and` versus `&&`.

Reply:
```python
>>> from time import sleep
>>> from timeit import timeit
>>> a = lambda: False
>>> def b():
...  sleep(4)
...  return True
...
>>> def c():
...  sleep(2)
...  return True
...
>>> def test():
...  if a() and b() and c():
...   pass
...
>>> timeit(test, number=1)
4.37488779425621e-06
```

Reply:
😂 I see what you mean, but here's a version that might be clearer for others:

```python
>>> def test(msg, ret):
...     print(msg)
...     return ret
...
>>> if test('first', True) and test('second', False) and test('third', False):
...     print('done')
...
first
second
>>>
```